src/runtime/current_thread/runtime.rs

```diff
@@ -42,12 +44,32 @@ impl Runtime {
         // And now put a single-threaded executor on top of the timer. When there are no futures ready
         // to do something, it'll let the timer or the reactor to generate some new stimuli for the
         // futures to continue in their life.
-        let executor = CurrentThread::new_with_park(timer);
+        let mut executor = CurrentThread::new_with_park(timer);
 
-        let runtime = Runtime { reactor_handle, timer_handle, executor };
+        // Create an mpsc channel for scheduling new tasks on this runtime
+        // from a different thread
+        // XXX: Should this use a bounded channel with an arbitrary limit instead, and then
+        // panic spawning if the channel is full?
```

Comment:
Are there any opinions on this?

Also some further work is needed so that `run()` ignores our channel instead of just running forever

Reply:
... let's just move it into the executor directly. There there's control over all that :)